ssl/ssl_cert.cc

```diff
@@ -153,20 +155,26 @@ UniquePtr<CERT> ssl_cert_dup(CERT *cert) {
     return nullptr;
   }
 
-  if (cert->chain) {
-    ret->chain.reset(sk_CRYPTO_BUFFER_deep_copy(
-        cert->chain.get(), buffer_up_ref, CRYPTO_BUFFER_free));
-    if (!ret->chain) {
-      return nullptr;
+  for (int i = 0; i < SSL_PKEY_NUM; i++) {
+    CERT_PKEY &cert_pkey = cert->cert_privatekeys[i];
+    CERT_PKEY &ret_pkey = ret->cert_privatekeys[i];
```

Comment:
There should be a check on the `size()` prior to iterating -- either `cert`  or `ret` might not have properly allocated arrays.